test/tstQueryTreeExactNearestDistance.cpp

```diff
@@ -0,0 +1,149 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <boost/test/unit_test.hpp>
+
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <tuple>
+
+#include "Search_UnitTestHelpers.hpp"
+// clang-format off
+#include "ArborXTest_TreeTypeTraits.hpp"
+// clang-format on
+#include <ArborX_Ray.hpp> // Vector
+
+namespace tt = boost::test_tools;
+
+struct Segment
+{
+  ArborX::Point start;
+  ArborX::Point end;
+};
+
+template <typename MemorySpace>
+struct Segments
+{
+  Kokkos::View<Segment *, MemorySpace> segments;
+};
+
+template <typename MemorySpace>
+struct ArborX::AccessTraits<Segments<MemorySpace>, ArborX::PrimitivesTag>
+{
+  using memory_space = MemorySpace;
+  using Primitives = Segments<MemorySpace>;
+
+  static KOKKOS_FUNCTION std::size_t size(Primitives const &primitives)
+  {
+    return primitives.segments.extent(0);
+  }
+  static KOKKOS_FUNCTION ArborX::Box get(Primitives const &primitives,
+                                         std::size_t i)
+  {
+    using namespace ArborX::Details;
+
+    ArborX::Box box;
+    expand(box, primitives.segments(i).start);
+    expand(box, primitives.segments(i).end);
+    return box;
+  }
+};
+
+// distance point-segment
+KOKKOS_INLINE_FUNCTION
+float distance(ArborX::Point const &point, Segment const &segment)
+{
+  using namespace ArborX::Experimental;
+  using namespace ArborX::Details;
+
+  Vector const dir = makeVector(segment.start, segment.end);
+  Vector const pa = makeVector(segment.start, point);
+  float const scale = dotProduct(pa, dir) / dotProduct(dir, dir);
+  ArborX::Point projection;
+  for (int d = 0; d < 3; ++d)
+    projection[d] = pa[d] - scale * dir[d];
+
+  float distance2start = distance(point, segment.start);
+  float distance2end = distance(point, segment.end);
+  float distance2projection = distance(point, projection);
+
+  using KokkosExt::min;
+  return min(distance2projection, min(distance2start, distance2end));
```

Comment:
```suggestion
  return min({distance2projection, distance2start, distance2end});
```

Reply:
But I don't think that is what you wanted if `scale < 0` or `scale > 1`

Reply:
You are right. I don't know why I had this stupidity in my head.